src/main/java/lotto/domain/Lotto.java

```diff
@@ -1,54 +1,104 @@
 package lotto.domain;
 
+import lotto.view.InputView;
+
 import java.util.*;
 
 public class Lotto {
 
-	public static final int LOTTO_PRICE_PER_SHEET = 1000;
-
 	private static final int LOTTO_NUMBER_MAX_COUNT = 6;
 
 	private static final int TOTAL_LOTTO_NUMBER_MAX = 45;
 
+	private List<Integer> lottoNumbers = new ArrayList<>();
+
 	private static List<Integer> defaultLottoNumbers = new ArrayList<>();
 
-	public Lotto() {
+	static {
 		for (int i = 0; i < TOTAL_LOTTO_NUMBER_MAX; i++) {
-			this.defaultLottoNumbers.add(i + 1);
+			defaultLottoNumbers.add(i + 1);
 		}
 	}
 
-	public List<List<Integer>> generateAutoLottoNumbers(int lottoCount) {
-		List<List<Integer>> autoGeneratedLottos = new ArrayList<>();
-		for (int i = 0; i < lottoCount; i++) {
-			List<Integer> lottoNumber = generateLotto();
-			autoGeneratedLottos.add(lottoNumber);
+	public Lotto() {
+	}
+
+	public Lotto(List<Integer> inputNumber) {
+		for (int number : inputNumber) {
+			lottoNumbers.add(number);
 		}
-		return autoGeneratedLottos;
+		validateMaxCount(lottoNumbers);
 	}
 
-	private List<Integer> generateLotto() {
+	public List<Integer> getLottoNumbers() {
+		return lottoNumbers;
+	}
+
+	public List<Integer> getShuffleNumbers() {
 		Collections.shuffle(defaultLottoNumbers);
-		List<Integer> lottoNumber = new ArrayList<>();
+		List<Integer> autoLottos = new ArrayList<>();
 		for (int i = 0; i < LOTTO_NUMBER_MAX_COUNT; i++) {
-			lottoNumber.add(defaultLottoNumbers.get(i));
+			autoLottos.add(defaultLottoNumbers.get(i));
+		}
+		return autoLottos;
+	}
+
+	public List<Lotto> generateAutoLottoNumbers(int lottoPayCount) {
+		List<Lotto> generatedLottos = new ArrayList<>();
+		for (int i = 0; i < lottoPayCount; i++) {
+			generatedLottos.add(new Lotto(this.getShuffleNumbers()));
+		}
+		return generatedLottos;
+	}
+
+	public int getMatchCount(Lotto lastWeekPrizeNumberList, Lotto generatedLotto) {
+		return countMatchNumber(lastWeekPrizeNumberList, generatedLotto);
+	}
+
+	public List<Lotto> generatePurchagedAutoLotto(int lottoTryCount) {
+		InputView.printPurchaseLottoCount(lottoTryCount);
+		List<Lotto> generatedLottos = new ArrayList<>();
+		for (int i = 0; i < lottoTryCount; i++) {
+			generatedLottos.add(new Lotto(this.getShuffleNumbers()));
+			generatedLottos.get(i).toString();
+		}
+
+		return generatedLottos;
+	}
+
+	public int countMatchNumber(Lotto lastWeekPrizeNumberList, Lotto generatedLotto) {
+		return lastWeekPrizeNumberList.countMatchNumber(generatedLotto);
+	}
+
+	public int countMatchNumber(Lotto generatedLotto) {
```

Comment:
```suggestion
		public int countMatchNumber(Lotto generatedLotto) {
		return (int) this.lottoNumbers.stream()
				.filter(number -> generatedLotto.lottoNumbers.contains(number))
				.count();
	}
```

코드가 많으면 내용 파악이 어렵고 유지보수할 때 신경써야 할 부분이 많아집니다.
코드량을 어떻게 하면 줄일 수 있을지 고민해보는게 좋겠네요